src/app/shared/services/app.service.ts

```diff
@@ -638,6 +650,25 @@ export class NSSService {
             .get(this.configSettings.gageStatsBaseURL + this.configSettings.stationsURL + "?stationTypes=" + id + "?agencies=" + id2)
     }
```

Comment:
Since you're combining your filters in the `searchStations` function, you can remove the `getStationsByType` and `getStationsByAgency` functions